### PR TITLE
Adds handler interface for property value resolution

### PR DESCRIPTION
**Pull request by justiandre**
Property value resolution in config.Provider now passes through a new interface config.ValueResolver, check documentation for more information.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +33,22 @@ if cfg, err = loader.Load(); err != nil {
 value := cfg.String("key")
 ```
 
+## Custom Property Recovery
+
+It is possible to customize the recovery of properties by name, to recover for example ignoring the camel case.
```
Comment (andrealbinop) [inline README.md]: It's possible
```suggestion
You can customize `config.Provider` property recovering routine, for instance, ignore camel case, as sampled below:
```

File: pkg/config/config.go [hunk 2]
```diff
@@ -62,3 +62,11 @@ type Loader interface {
 	// Load the config.Provider
 	Load() (Provider, error)
 }
+
+// NameResolver interface to customize property recovery by name
+type NameResolver interface {
+	// Lookup returns a value associated with the name.
+	Lookup(string, Repository) (interface{}, bool)
```
Comment (andrealbinop) [inline pkg/config/config.go]:
```suggestion
	Resolve(string, Repository) (interface{}, bool)
```

File: pkg/nameresolver/nameresolver_test.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+func createCfgProviderDefaultNameResolver() provider.Default {
+	return provider.Default{
+		Repository:   &repository.Map{},
+		NameResolver: &Default{},
+	}
+}
+
+type NameResolverRelaxedName struct {
```
Comment (andrealbinop) [inline pkg/nameresolver/nameresolver_test.go]: Are you sure we need to test this here?

File: pkg/config/config.go [hunk 4]
```diff
@@ -62,3 +62,11 @@ type Loader interface {
 	// Load the config.Provider
 	Load() (Provider, error)
 }
+
+// NameResolver interface to customize property recovery by name
+type NameResolver interface {
```
Comment (andrealbinop) [inline pkg/config/config.go]: Name is misleading in my opinion, you're actually resolving property values, not names...
```suggestion
type ValueResolver interface {
```

**Review by andrealbinop — CHANGES_REQUESTED** [review]
Suggestions regarding commit/pr:

> Adds handler interface for property value resolution

Property value resolution in `config.Provider` now passes through a new interface `config.NameResolver`, check documentation for more information

File: pkg/nameresolver/nameresolver_test.go (hunk 3 above)
Comment (justiandre) [reply pkg/nameresolver/nameresolver_test.go]: Not necessary. But it's a good case of being tested.

**Review by andrealbinop — APPROVED** [review]

**Pull request merged**